include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm.hpp

```diff
@@ -141,11 +141,10 @@ class pairwise_alignment_algorithm : protected policies_t...
     {
         using std::get;
 
-        for (auto && indexed_sequence_pair : indexed_sequence_pairs)
+        for (auto && [sequence_pair, idx] indexed_sequence_pair : indexed_sequence_pairs)
         {
-            auto && sequence_pair = get<0>(indexed_sequence_pair);
             compute_matrix(get<0>(sequence_pair), get<1>(sequence_pair));
-            this->make_result_and_invoke(indexed_sequence_pair, this->tracked_optimum(), callback);
+            this->make_result_and_invoke(sequence_pair, std::move(idx), this->tracked_optimum(), callback);
```

Comment:
```suggestion
            this->make_result_and_invoke(std::forward<decltype(sequence_pair)>(sequence_pair), std::move(idx), this->tracked_optimum(), callback);
```

Does this make sense? I'm not sure

Reply:
Huh, I need to think about it.

Reply:
Is this still open or is there a resolution?